### PR TITLE
Include deprecated Gallery block in tests.

### PR DESCRIPTION
**Pull request by peterwilsoncc**
Follow up to 3cc94be23b31d8eb22c81a615b749f82bd6de2ad.

This includes the deprecated gallery blocks in the core unit tests. This is to ensure the blocks render as expected in their deprecated forms in both the plugin and once merged in to the WordPress-Develop repo.

`no-idea.gif` so initially this is just throwing something messy at the wall and hoping it sticks.

https://core.trac.wordpress.org/ticket/55571

### CODE REVIEW

File: tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-6.server.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+				<a
+					href="http://wptest.local/wp-content/uploads/2020/04/test-image-scaled.jpg"
+				>
+					<img
+						src="http://wptest.local/wp-content/uploads/2020/04/test-image-1024x683.jpg"
+						alt=""
+						data-id="703"
+						data-full-url="http://wptest.local/wp-content/uploads/2020/04/test-image-scaled.jpg"
+						data-link="http://wptest.local/test-image/"
+						class="wp-image-703 is-layout-flex"
```
Comment (peterwilsoncc) [inline tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-6.server.html]: `is-layout-flex` on this doesn't seem right to me. I'll check it again an old branch.

It's applied to this element but none of the other images, not is it applied at the root gallery.

File: tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-2.server.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+<ul class="wp-block-gallery columns-2 is-cropped alignwide wp-block-gallery-1">
+	<li class="blocks-gallery-item">
+		<figure>
+			<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAACklEQVR4nGMAAQAABQABDQottAAAAABJRU5ErkJggg==" data-id="1" alt="title" class="wp-image-1" />
+		</figure>
+	</li>
+	<li class="blocks-gallery-item">
+		<figure>
+			<img src="data:image/jpeg;base64,/9j/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/yQALCAABAAEBAREA/8wABgAQEAX/2gAIAQEAAD8A0s8g/9k=" data-id="2" alt="title" class="wp-image-2 is-layout-flex" />
```
Comment (peterwilsoncc) [inline tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-2.server.html]: `is-layout-flex` doesn't look appropriate here as it's not on the other images. I am unsure if it has an effect.

File: tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-4.server.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+			</figure>
+		</li>
+		<li class="blocks-gallery-item">
+			<figure>
+				<img src="https://sergioestevaofolio.files.wordpress.com/2017/09/cropped-l1001498-1.jpg?w=580" alt="" data-id="1440" class="wp-image-1440"/>
+			</figure>
+		</li>
+		<li class="blocks-gallery-item">
+			<figure>
+				<img src="https://sergioestevaofolio.files.wordpress.com/2017/05/cropped-l1005945-2-2.jpg?w=580" alt="" data-id="1362" class="wp-image-1362 is-layout-flex"/>
```
Comment (peterwilsoncc) [inline tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-4.server.html]: `is-layout-flex` doesn't look appropriate here as it's not on the other images. I am unsure if it has an effect.

File: tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-5.server.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		<li class="blocks-gallery-item">
+			<figure>
+				<a href="http://wptest.local/wp-content/uploads/2020/04/test-image-scaled.jpg" >
+					<img
+						src="http://wptest.local/wp-content/uploads/2020/04/test-image-1024x683.jpg"
+						alt=""
+						data-id="703"
+						data-full-url="http://wptest.local/wp-content/uploads/2020/04/test-image-scaled.jpg"
+						data-link="http://wptest.local/test-image/"
+						class="wp-image-703 is-layout-flex"
```
Comment (peterwilsoncc) [inline tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-5.server.html]: `is-layout-flex` doesn't look appropriate here as it's not on the other images. I am unsure if it has an effect.

**Review by peterwilsoncc — COMMENTED** [review]
At time of writing this just fixes the `test_do_block_output` a little more work to do with the parsed json version.

File: tests/phpunit/tests/blocks/render.php [hunk 5]
```diff
@@ -233,6 +233,10 @@ public function test_do_block_output( $html_filename, $server_html_filename ) {
 		$normalized_html = preg_replace( '/wp-block-gallery-\d+/', 'wp-block-gallery-1', $normalized_html );
 		$expected_html   = self::strip_r( file_get_contents( $server_html_path ) );
 
+		// Convert HTML to be white space insensitive.
+		$normalized_html = preg_replace( '/(\s+$)/m', '', $normalized_html );
+		$expected_html   = preg_replace( '/(\s+$)/m', '', $expected_html );
+
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/blocks/render.php:233]: This change strips whitespace from the end of lines in both the normalized and expected HTML.

When rendering the blocks, Gutenberg often adds a white space following any dynamic attributes, eg `data-id="1234" ` which is correct but if it comes in at the end of the line, can be missed in the expected version.

As HTML treats one or more instances of white space identically, it doesn't have an effect on the behavior in browsers.

File: tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-1.parsed.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+		},
+		"innerBlocks": [],
+		"innerHTML": "\n<div class=\"wp-block-gallery columns-2 is-cropped alignwide\">\n\t<figure class=\"blocks-gallery-image\">\n\t\t<img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAACklEQVR4nGMAAQAABQABDQottAAAAABJRU5ErkJggg==\" alt=\"title\" />\n\t</figure>\n\t<figure class=\"blocks-gallery-image\">\n\t\t<img src=\"data:image/jpeg;base64,/9j/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/yQALCAABAAEBAREA/8wABgAQEAX/2gAIAQEAAD8A0s8g/9k=\" alt=\"title\" />\n\t</figure>\n</div>\n",
+		"innerContent": [
+			"\n<div class=\"wp-block-gallery columns-2 is-cropped alignwide\">\n\t<figure class=\"blocks-gallery-image\">\n\t\t<img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAACklEQVR4nGMAAQAABQABDQottAAAAABJRU5ErkJggg==\" alt=\"title\" />\n\t</figure>\n\t<figure class=\"blocks-gallery-image\">\n\t\t<img src=\"data:image/jpeg;base64,/9j/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/yQALCAABAAEBAREA/8wABgAQEAX/2gAIAQEAAD8A0s8g/9k=\" alt=\"title\" />\n\t</figure>\n</div>\n"
+		]
+	},
+	{
+		"blockName": null,
```
Comment (tellthemachines) [inline tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-1.parsed.json:15]: Weird, I wonder where these null blocks come from?

**Review by tellthemachines — COMMENTED** [review]
I'm not 100% sure what's happening with the null blocks in some of the parsed JSON files, though I notice there's a few in Gutenberg too. @gziolo any idea what might be going on here?

The classnames are behaving as expected though! And the PHP tests are passing locally ✅

**Comment by peterwilsoncc** [comment]
I think the null blocks come from end of file empty lines. 
I tried using trim to remove them but it broke a bunch of other tests that already had trailing null blocks recorded.

File: tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-1.parsed.json (hunk 6 above)
Comment (gziolo) [reply tests/phpunit/data/blocks/fixtures/core__gallery__deprecated-1.parsed.json:15]: I also believe this comes from empty lines like Pete mentioned in https://github.com/WordPress/wordpress-develop/pull/4140#issuecomment-1451472711.

**Review by tellthemachines — APPROVED** [review]
LGTM!

**Comment by peterwilsoncc** [comment]
Merged in [r55471](https://core.trac.wordpress.org/changeset/55471) / https://github.com/WordPress/wordpress-develop/commit/4d3627fd2eab274c55ca4fb22dbdc0de64b17112

**Pull request closed without merging**